easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,227 @@
+
+        if self.cfg['with_shared']:
+            self.cfg.update('configopts', '--enable-shared')
+
+        if self.cfg['with_generic_kernel']:
+            self.cfg.update('configopts', '--enable-generic')
+
+        if self.cfg['with_single']:
+            self.cfg.update('configopts', '--enable-single-precision')
+
+        for flag in ELPA_CPU_FEATURE_FLAGS:
+            # many ELPA kernels are enabled by default, even when the
+            # CPU does not support them, so we disable them all, except
+            # when the appropriate CPU flag is found
+            # sse kernels require sse4_2
+            if flag == 'sse4_2':
+                if getattr(self, flag):
+                    self.cfg.update('configopts','--enable-sse')
+                    self.cfg.update('configopts', '--enable-sse-assembly')
+                else:
+                    self.cfg.update('configopts', '--disable-sse')
+                    self.cfg.update('configopts', '--disable-sse-assembly')
+            elif flag == 'avx512f':
+                if getattr(self, 'avx512f'):
+                    self.cfg.update('configopts', '--enable-avx512')
+                else:
+                    self.cfg.update('configopts', '--disable-avx512')
+            else:
+                if getattr(self, flag):
+                    self.cfg.update('configopts', '--enable-%s' % flag)
+                else:
+                    self.cfg.update('configopts', '--disable-%s' % flag)
+
+        # By default ELPA tries to use MPI and configure fails if it's not available
+        # so we turn it off if MPI is not available
+        # We also set the LIBS environmet variable to detect the correct linalg library
+        # depending on the MPI availability.
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', 'LIBS="$LIBSCALAPACK"')
+        else:
+            self.cfg.update('configopts', '--with-mpi=no')
+            self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
+
+        # make all builds verbose
+        self.cfg.update('buildopts', 'V=1')
+
+        # keep track of common configopts and of configopts specified
+        # in easyconfig file, so we can include them in each iteration
+        # later
+        common_config_opts = self.cfg['configopts']
+        common_build_opts = self.cfg['buildopts']
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
+
+        with_omp_opts = [False]
+        if self.cfg['with_openmp']:
+            with_omp_opts.append(True)
+
+        for with_omp in with_omp_opts:
+            if with_omp:
+                omp_configopt = '--enable-openmp'
+            else:
+                omp_configopt = '--disable-openmp'
+
+            # append additional configure and build options
+            self.cfg.update('configopts',
+                            [omp_configopt + ' ' + common_config_opts])
+            self.cfg.update('buildopts', [common_build_opts])
```

Comment:
This can be simplified imho, since there can only be two values in `with_omp_opts`:

```python
self.cfg.update('configopts', ['--disable-openmp ' + common_config_opts])

if self.toolchain.options.get('openmp', False):
    self.cfg.update('configopts', ['--enable-openmp ' + common_config_opts])